rest/views.py

```diff
@@ -132,6 +133,40 @@ def get_queryset(self):
         return queryset.order_by('id')
 
 
+class CourseDetail(RetrieveDestroyAPIView):
+    """REST detail for Course"""
+    serializer_class = CourseSerializer
+    permission_classes = (
+        ViewRepoPermission,
+        ImportCoursePermission,
+        IsAuthenticated,
+    )
+
+    def get_object(self):
+        """
+        Return details about a course in a repo
+        """
+        try:
+            course = Course.objects.get(id=int(self.kwargs['course_id']))
+        except (Course.DoesNotExist, ValueError):
+            raise Http404()
+        try:
+            assert course.repository.slug == self.kwargs['repo_slug']
```

Comment:
We shouldn't be using `assert` for checks in production, this could theoretically be skipped if the server ran `python -O`, and from a clean code point of view asserts verify something we know to be true but assert to be on the safe side. Since this could reasonably be false it should just be an if statement instead.